arkane/ess/adapter.py

```diff
@@ -28,103 +29,116 @@
 ###############################################################################
 
 """
-A general class for parsing quantum mechanical log files
+A module for the abstract ESSAdapter class
 """
 
+from abc import ABC, abstractmethod
 import logging
-import os.path
+import os
 import shutil
-from abc import ABC, abstractmethod
 
 from rmgpy.qm.qmdata import QMData
 from rmgpy.qm.symmetry import PointGroupCalculator
 
-################################################################################
-
-
-class Log(ABC):
+class ESSAdapter(ABC):
     """
-    Represent a general log file.
-    The attribute `path` refers to the location on disk of the log file of interest.
+    An abstract ESS Adapter class
     """
 
     @abstractmethod
     def get_number_of_atoms(self):
         """
         Return the number of atoms in the molecular configuration.
-        Should be implemented by a subclass.
         """
         pass
 
     @abstractmethod
     def load_force_constant_matrix(self):
         """
-        Return the force constant matrix (in Cartesian coordinates).
-        Should be implemented by a subclass.
+        Return the force constant matrix (in Cartesian coordinates). If multiple such matrices
+        are identified, only the last is returned. The units of the returned force constants
+        are J/m^2. If no force constant matrix can be found in the log file, ``None`` is returned.
         """
         pass
 
     @abstractmethod
     def load_geometry(self):
         """
         Return the optimum geometry of the molecular configuration.
-        Should be implemented by a subclass.
+        If multiple such geometries are identified, only the last is returned.
         """
         pass
 
     @abstractmethod
-    def load_conformer(self, symmetry=None, spin_multiplicity=0, optical_isomers=None, label=''):
```

Comment:
It appears that your suggestion would be legal! I wasn't expecting that. I thought we had to define the arguments for this method in Log just in case a piece of code tried to pass non-default arguments to this method..

Reply:
I think we should keep the arguments in the abstract method call, unless it's problematic or I missed something

Reply:
Resolved via our zoom call. The arguments are now present in the abstract method